### DISCUSSION TITLE
Other Aggregate functions

### DISCUSSION THREAD

**Discussion by ashgibson** [spatie/laravel-query-builder#799] [Ideas]
Any plans to implement the other aggregate functions in the allowedIncludes?

https://laravel.com/docs/9.x/eloquent-relationships#other-aggregate-functions

**Comment by freekmurze** [comment]
We have no plans for that, but feel free to submit PRs.

**Comment by freekmurze** [comment]
This is coming in the next major version (v7) of the package.

You will be able to use aggregate includes like this:

```php
QueryBuilder::for(User::class)
    ->allowedIncludes([
        AllowedInclude::min("postsViewsMin", "posts", "views"),
        AllowedInclude::max("postsViewsMax", "posts", "views"),
        AllowedInclude::sum("postsViewsSum", "posts", "views"),
        AllowedInclude::avg("postsViewsAvg", "posts", "views"),
    ]);
```

The suffixes (`Min`, `Max`, `Sum`, `Avg`) are configurable in the `query-builder` config file.

Docs will be available at https://spatie.be/docs/laravel-query-builder/v7/features/including-relationships once v7 is released.
